src/runtime/offers-flow.js

```diff
@@ -51,20 +52,54 @@ export class OffersFlow {
       isClosable = false; // Default is to hide Close button.
     }
 
+    const feArgsObj = {
+      'productId': deps.pageConfig().getProductId(),
+      'publicationId': deps.pageConfig().getPublicationId(),
+      'showNative': deps.callbacks().hasSubscribeRequestCallback(),
+      'productType': ProductType.SUBSCRIPTION,
+      'list': (options && options.list) || 'default',
+      'skus': (options && options.skus) || null,
+      'isClosable': isClosable,
+    };
+
+    if (options && options.oldSku) {
+      feArgsObj['oldSku'] = options.oldSku;
+    }
+
+    if (feArgsObj['oldSku']) {
+      assert(feArgsObj['skus'], 'Need a sku list if old sku is provided!')
+      // remove old sku from offers if in list
+      let skuList = feArgsObj['skus'];
+      const /** @type {String} */ oldSku = feArgsObj['oldSku'];
+      skuList = skuList.filter(sku => sku !== oldSku);
+      assert(skuList.length > 0, 'Sku list only contained offer user already has');
+      feArgsObj['skus'] = skuList;
+    }
+
+    // redirect to payments if only one upgrade option is passed
+    if (feArgsObj['skus'] && feArgsObj['skus'].length === 1) {
+      const sku = feArgsObj['skus'][0];
+      const /** @type {String} */ oldSku = feArgsObj['oldSku'];
+      // object currently requires experimental flag
+      // so we need to check for oldSku to decide what to send
+      if (oldSku) {
+        new PayStartFlow(this.deps_, {
+          skuId: sku,
+          oldSkuId: oldSku,
+          replaceSkuProrationMode: ReplaceSkuProrationMode.IMMEDIATE_WITH_TIME_PRORATION,
+        }).start();
+        return;
+      } else {
+        new PayStartFlow(this.deps_, sku).start();
```

Comment:
As discussed offline, this `else` clause requires some discussion.

showOffers() can be launched by a publisher without user interaction and for a single SKU, launching a `PlayFlow` seems odd to me (and may be for the users too).

Reply:
I've commented out the auto-redirect for when there's no oldSku. (Actually I probably should have deleted the code outright - I'll fix that in the next push.)